### ISSUE TITLE
Rubocop not found if not in global PATH variable

### ISSUE THREAD

**Issue by cschneid** [yujinakayama/atom-lint#30]
I am launching atom from a terminal that has RVM installed, in a gemset with Rubocop available.

```
$ which rubocop
/Users/cschneid/Developer/.rvm/gems/ruby-2.1.1@advanced-apps/bin/rubocop

$ atom .
```

The atom-lint package cannot find the rubocop executable, probably due to some of the PATH hackery done in this commit: https://github.com/yujinakayama/atom-lint/commit/7702e76f124ccbf1289577262d9441b7e246ce11

I like the goal of that commit, and I can work around this issue by manually configuring the path to rubocop, but I had expected that it would load the path that I had when I launched the editor.

Maybe support both?

Thanks for the otherwise awesome tool! Once I had the workaround to set the path, the rest is looking really cool.

**Comment by kemelzaidan** [comment]
I'm having the same problem with rbenv:

```
$ which rubocop
/Users/kemelzaidan/.rbenv/shims/rubocop
```

```
process.env.PATH
"/usr/bin:/bin:/usr/sbin:/sbin"
```

rbenv config in bash:

```
export PATH="$HOME/.rbenv/bin:$PATH"
eval "$(rbenv init -)"
```

Error:
The linter binary 'rubocop' cannot be found.
Linting has been halted. Please install the linter binary or disable the linter plugin depending on it. Make sure to reload Atom to detect changes
Close All
Failed to spawn command rubocop. Make sure rubocop is installed and on your PATH

**Comment by mmahalwy** [comment]
I have the same problem @kemelzaidan but with fish shell. Any luck?

**Comment by kWhittington** [comment]
Same for me. Atom's actually brining up the developer console too and I can't close it.
- fish, version 2.1.1
- rvm 1.26.11 (latest) by Wayne E. Seguin wayneeseguin@gmail.com, Michal Papis mpapis@gmail.com [https://rvm.io/]
- atom -v 1.3.1

**Comment by bbrock25** [comment]
I am having similar issues. The best I've found is to specify `bundle exec rubocop` as the custom command. However it still chokes when I specify the full path for bundler.

**Comment by ilyakatz** [comment]
Having the same issue, anyone has any ideas?

**Comment by sergiotapia** [comment]
This is still happening :(

**Comment by andretf** [comment]
I'm using **rbenv**, using shims path does not work. Set path to `~/.rbenv/versions/2.3.1/bin/rubocop` and it should work.

**Comment by graciano** [comment]
In case anyone ran into this issue with the same problem that I had:
The project I'm working on uses a legacy version of ruby: 1.9, and this is not compatible with the current version of the gem rubocop. So installing a compatible version worked for me
```shell
gem install rubocop -v 0.41.2
```

**Comment by compostbrain** [comment]
I have the same issue as @andretf .  How do I set the path to `~/.rbenv/versions/2.3.1/bin/rubocop`?

**Comment by merefield** [comment]
thanks @graciano!

`gem install rubocop`

Was enough for me to fix this.

**Comment by wbotelhos** [comment]
Thank you @yujinakayama ,

After months not using this plugin, even trying a lot of thing, now I know the differents env depending of the way you open the app (Atom in my case).

**Comment by clockworkpc** [comment]
> I think the issue is fixed in version 0.8.1.
> 
> Please note that if you've launched Atom from Finder/Dock _first_ and then opened another project by running `atom` command in your shell, the Atom process _always_ uses the environment variables of Finder/Dock even in the project window launched from shell. This is unavoidable and cannot be handled in atom-lint.
> 
> If you still have the issue, please reopen.

HALLELUJAH.
This fixed it for me.
